clientv3/watch.go

```diff
@@ -333,7 +334,7 @@ func (w *watcher) Watch(ctx context.Context, key string, opts ...OpOption) Watch
 	case <-wr.ctx.Done():
 	case <-donec:
 		if wgs.closeErr != nil {
-			closeCh <- WatchResponse{closeErr: wgs.closeErr}
+			closeCh <- WatchResponse{Canceled: true, closeErr: wgs.closeErr}
```

Comment:
@xiang90 This changes the behavior of watch cancel? I will check if it would break other projects before v3.4 release.

Reply:
@gyuho I do not think so. When closeErr is not nil, we should always set Canceled to true.

Reply:
I am not sure why we did not do it before.

Reply:
@xiang90 

> I am not sure why we did not do it before.

Yeah, we considered the same change as we were writing v3.4 balancer, but decided not to. 

Couldn't remember why we have this in the first place.
I will be happy to ship this in v3.4 as a bug fix :)

Will double-check before the release.

/cc @jpbetz